pallets/omnipool/src/lib.rs

```diff
@@ -1394,6 +1450,14 @@ impl<T: Config> Pallet<T> {
 			asset_out_state.tradable.contains(Tradability::BUY),
 			Error::<T>::NotAllowed
 		);
+		ensure!(
+			amount
```

Comment:
Why do we compare `amount_in` with `asset_out_state`? Probably wrong name.

Reply:
it is actually ok because we work only with one asset state here. I renamed it to asset_state.